tests/modeladmin/tests.py

```diff
@@ -175,13 +188,70 @@ class WaiterAdmin(ModelAdmin):
             ]
 
         ma = WaiterAdmin(Waiter, self.site)
-        self.assertIs(ma.lookup_allowed("restaurant__place__country", "1"), True)
+        mock_request = MockRequest()
+        self.assertIs(
+            ma.lookup_allowed("restaurant__place__country", "1", request=mock_request),
+            True,
+        )
         self.assertIs(
-            ma.lookup_allowed("restaurant__place__country__id__exact", "1"), True
+            ma.lookup_allowed(
+                "restaurant__place__country__id__exact", "1", request=mock_request
+            ),
+            True,
         )
         self.assertIs(
-            ma.lookup_allowed("restaurant__place__country__name", "test_value"), True
+            ma.lookup_allowed(
+                "restaurant__place__country__name", "test_value", request=mock_request
+            ),
+            True,
+        )
+
+    def test_lookup_allowed_considers_dynamic_list_filter(self):
+        class ConcertAdmin(ModelAdmin):
+            list_filter = ["main_band__sign_date"]
+
+            def get_list_filter(self, request):
+                return self.list_filter + ["main_band__name"]
+
+        model_admin = ConcertAdmin(Concert, self.site)
+        request_band_name_filter = RequestFactory().get(
+            "/", {"main_band__name": "test"}
+        )
+        self.assertTrue(
```

Comment:
Following the code above and the [coding style guidelines](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style), I believe these should be `assertIs` instead of `assertTrue`.